tests/tacacs/test_ro_user.py

```diff
@@ -63,22 +77,25 @@ def test_ro_user_allowed_command(localhost, duthosts, rand_one_dut_hostname, cre
             'sudo generate_dump',
             'sudo lldpshow',
             # 'sudo psuutil *',
-            'sudo sonic-installer list',
             # 'sudo sfputil show *',
             'sudo ip netns identify 1',
     ]
-
-    # Run as readonly use the commands allowed indirectly based on sudoers file
+        # Run as readonly use the commands allowed indirectly based on sudoers file
     commands_indirect = [
             'show version',
     ]
 
     for command in commands_direct + commands_indirect:
-        res = ssh_remote_run(localhost, dutip, creds['tacacs_ro_user'], creds['tacacs_ro_user_passwd'], command)
-        # Verify that the command is allowed
-        logger.info("check command \"{}\" rc={}".format(command, res['rc']))
-        pytest_assert(res['rc'] == 0 or (res['rc'] != 0 and "Make sure your account has RW permission to current device" not in res['stderr']),
-                "command '{}' not authorized".format(command))
+        allowed = ssh_remote_allow_run(localhost, dutip, creds['tacacs_ro_user'], creds['tacacs_ro_user_passwd'], command)
+        pytest_assert(allowed, "command '{}' not authorized".format(command))
+
+    dash_allowed = ssh_remote_allow_run(localhost, dutip, creds['tacacs_ro_user'], creds['tacacs_ro_user_passwd'], 'sudo sonic-installer list')
```

Comment:
@qi-luo may be we can make this generic of another list of commands that can have dash/under-score. Then we can loop through all of those commands so that it can extendible for any same case going forward.

Reply:
we can do it in future. this is a bug fix, so to keep it as simple as possible